ipa-core/src/ff/boolean_array.rs

```diff
@@ -634,6 +648,12 @@ macro_rules! boolean_array_impl {
                     }
                 }
 
+                #[test]
+                fn signed_integer_conversions() {
```

Comment:
I suggest you test conversion for 0 and MAX values as well. If you plan to keep it in this module, I think we should have this test for all fields using $name